javalib/src/main/scala/java/util/Base64.scala

```diff
@@ -387,4 +453,34 @@ object Base64 {
       }
     }
   }
+
+  /** hides index manipulation operations */
+  private class Wrapper[@specialized(Byte, Int) T](array: Array[T], var position: Int, var length: Int) {
+
+    def this(array: Array[T]) = this(array, 0, array.length)
+
+    def hasRemaining: Boolean = position < length
+
+    def remaining: Int = length - position
+
+    def put(t: T): Unit = {
+      array(position) = t
+      position += 1
+    }
+
+    def get: T = {
+      position += 1
+      array(position-1)
+    }
+
+    def clear(): Unit = {
+      position = 0
+      length = array.length
+    }
+
+    def flip(): Unit = {
+      length = position
+      position = 0
+    }
+  }
```

Comment:
Ok so maybe we'll have to agree to disagree on what's readable, but I find it better if the index/position manipulation operations on arrays are hidden in a wrapper like this, especially since a syntax like array(index++) is not possible. I hope the @specialized stuff makes sense, let me know if that's pointless.